src/Dialog.ts

```diff
@@ -17,45 +17,41 @@
 const {assert} = require('./helper');
 
 class Dialog {
-  /**
-   * @param {!Puppeteer.CDPSession} client
-   * @param {string} type
-   * @param {string} message
-   * @param {(string|undefined)} defaultValue
-   */
-  constructor(client, type, message, defaultValue = '') {
+
+  private _client: Puppeteer.CDPSession;
+  private _type: string
+  private _message: string
+  private _defaultValue: string
+
+  private _handled: boolean = false;
+
+  static Type = {
+    Alert: 'alert',
+    BeforeUnload: 'beforeunload',
+    Confirm: 'confirm',
+    Prompt: 'prompt'
+  }
```

Comment:
TODO: I think that `_type` is actually one of these, rather than just `string` as the old jsdoc comment declared